06_gpu_and_ml/stable_diffusion/stable_diffusion_cli.py

```diff
@@ -80,46 +84,39 @@
 # 1.6s per generation on average. On a T4, it takes 13s to load and 3.7s per
 # generation. Other optimizations are also available [here](https://huggingface.co/docs/diffusers/optimization/fp16#memory-and-speed).
 
-# This is our Modal function. The function runs through the `StableDiffusionPipeline` pipeline.
+# This is our Modal function. The function runs through the `StableDiffusion3Pipeline` pipeline.
 # It sends the PIL image back to our CLI where we save the resulting image in a local file.
 
 
-@app.cls(image=image, gpu="A10G")
+@app.cls(
+    image=image,
+    gpu="A100",
+    secrets=[modal.Secret.from_name("huggingface-secret-ren")],
```

Comment:
We did not previously use a Secret.

Could you poke around for a version of this model that doesn't require accepting a license? There are usually forks of popular models.